### PR TITLE
modify MarkdownHtmlBlockRule in markdown.

### PR DESCRIPTION
**Pull request by vwxyzh**

### CODE REVIEW

File: src/Microsoft.DocAsCode.MarkdownLite/Basic/BlockRules/MarkdownHtmlBlockRule.cs [hunk 1]
```diff
@@ -3,17 +3,166 @@
...
+
+        private static readonly Matcher _HtmlMatcher =
+            Matcher.WhiteSpacesOrEmpty +
+            (
+                // html comment
+                (
+                    Matcher.String("<!--") +
+                    (
+                        Matcher.AnyCharNot('-').RepeatAtLeast(1) |
+                        (Matcher.Char('-') + Matcher.String("->").ToNegativeTest())
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.MarkdownLite/Basic/BlockRules/MarkdownHtmlBlockRule.cs]: >(Matcher.Char('-') + Matcher.String("->") [](start = 24, length = 41)

what's the difference from Matcher.String("-->")?

**Comment by ansyral** [comment]
:shipit:

File: src/Microsoft.DocAsCode.MarkdownLite/Basic/BlockRules/MarkdownHtmlBlockRule.cs [hunk 2]
```diff
@@ -3,17 +3,166 @@
...
+                (
+                    Matcher.Char('<') +
+                    (
+                        (InlineElementNames + Matcher.AnyWordCharacter.ToNegativeTest()).ToNegativeTest() +
+                        // \w+
+                        Matcher.AnyWordCharacter.RepeatAtLeast(1) +
+                        // (?!:\/|[^\w\s@]*@)
+                        (
+                            Matcher.String(":/") |
+                            ((Matcher.AnyWordCharacter | Matcher.AnyCharIn(' ', '\n', '@')).ToNegativeTest() + Matcher.AnyChar).RepeatAtLeast(0) + '@'
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.MarkdownLite/Basic/BlockRules/MarkdownHtmlBlockRule.cs]: >Matcher.AnyChar [](start = 111, length = 15)

as `[^\w\s@]` matches `(..|..).ToNegativeTest` and `*` matches `RepeatAtLeast(0)`, why is `Matcher.AnyChar` needed?

Comment (vwxyzh) [inline src/Microsoft.DocAsCode.MarkdownLite/Basic/BlockRules/MarkdownHtmlBlockRule.cs]: `ToNegativeTest` is just test, it always return `0` (Ok) or `-1` (NG)

**Comment by vicancy** [comment]
:shipit:

**Comment by hellosnow** [comment]
:shipit:

File: src/Microsoft.DocAsCode.MarkdownLite/Basic/BlockRules/MarkdownHtmlBlockRule.cs [hunk 3]
```diff
@@ -3,17 +3,166 @@
...
+                        ).ToNegativeTest() +
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.MarkdownLite/Basic/BlockRules/MarkdownHtmlBlockRule.cs]: minor: need we extract the common code for (?!(?:inline\b)\w+(?!:\/|[^\w\s@]*@)\b?

**Pull request merged**